flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PartitionedFileReader.java

```diff
@@ -61,104 +56,69 @@
     /** Next data region to be read. */
     private int nextRegionToRead;
 
+    /** Next file offset to be read. */
+    private long nextOffsetToRead;
+
     /** Number of remaining buffers in the current data region read. */
     private int currentRegionRemainingBuffers;
 
-    /** Whether this partitioned file reader is closed. */
-    private boolean isClosed;
-
-    public PartitionedFileReader(PartitionedFile partitionedFile, int targetSubpartition)
+    public PartitionedFileReader(
+            PartitionedFile partitionedFile,
+            int targetSubpartition,
+            FileChannel dataFileChannel,
+            FileChannel indexFileChannel)
             throws IOException {
+        checkArgument(
```

Comment:
It's better to check them separately. And this check should be behind their `checkNotNull`.